### PR TITLE
Docs: Add linking topics

### PR DESCRIPTION
**Pull request by oddlittlebird**
This breaks up the Navigation links topic and moves the content from the Data links topic to URL variables. I also started moving files around to get closer to what the docs will look like for 7.0.

I dropped some content into a Sharing file, but it is currently all marked DRAFT, so should not show up. I need to do some more research before we finalize the linking/sharing/whatever content.

**Special notes for your reviewer**: Please read carefully. My brain is so fried right now, lol.

### CODE REVIEW

File: docs/sources/linking/linking-overview.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Which link should you use?
+
+Start by figuring out how you're currently navigating between dashboards. If you're often jumping between a set of dashboards and struggling to find the same context in each, links can help optimize your workflow.
+
+The next step is to figure out which link type is right for your workflow. Even though all the link types in Grafana are used to create shortcuts to other dashboards or external websites, they work in different contexts.
+
+- If the link relates to most if not all of the panels in the dashboard, use [dashboard links]({{< relref "dashboard-links.md" >}}).
+- If you want to drill down into specific panels, use [panel links]({{< relref "panel-links.md" >}}).
+- If you want to link to an external site, you can use either type of link.
```
Comment (dprokop) [inline docs/sources/linking/linking-overview.md:26]: We are missing data links here (https://grafana.com/docs/grafana/latest/reference/datalinks/)

File: docs/sources/variables/url-variables.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
++++
+title = "URL variables"
+keywords = ["grafana", "url variables", "documentation", "variables"]
+type = "docs"
++++
+
+# URL variables
+
+You can use variables in URLs to link to specific portions of your visualizations. For more information about linking, refer to [Dashboard links]({{< relref "../linking/dashboard-links.md" >}}) and [Panel links]({{< relref "../linking/panel-links.md" >}}).
```
Comment (dprokop) [inline docs/sources/variables/url-variables.md:9]: .... and Data links

File: docs/sources/variables/url-variables.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+{{< docs-imagebox img="/img/docs/data_link_typeahead.png"  max-width= "800px" >}}
+
+## Time variables
+
+* ``__url_time_range`` - current dashboard's time range (i.e. ``?from=now-6h&to=now``)
+* ``__from`` - current dashboard's time range from value
+* ``__to`` - current dashboard's time range to value
+
+## Series variables
```
Comment (dprokop) [inline docs/sources/variables/url-variables.md:26]: Available in Data links only

File: docs/sources/variables/url-variables.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+Series specific variables are available under ``__series`` namespace:
+
+* ``__series.name`` - series name to the URL
+* ``__series.labels.<LABEL>`` - label's value to the URL. If your label contains dots use ``__series.labels["<LABEL>"]`` syntax
+
+## Field variables
```
Comment (dprokop) [inline docs/sources/variables/url-variables.md:33]: Available in Data links only

File: docs/sources/variables/url-variables.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+Field specific variables are available under ``__field`` namespace:
+
+* ``__field.name`` - field name to the URL
+
+## Value variables
```
Comment (dprokop) [inline docs/sources/variables/url-variables.md:39]: Available in Data links only

**Review by dprokop — CHANGES_REQUESTED** [review]
Few changes suggested :)

**Comment by oddlittlebird** [comment]
> Few changes suggested :)

Yep, thanks for explaining where the data links are now. I'll get that content incorporated.

File: docs/sources/variables/url-variables.md (hunk 5 above)
Comment (oddlittlebird) [reply docs/sources/variables/url-variables.md:39]: Wait, what? The URL variables are only for data links? But you can add some variables to panel links.

**Comment by daniellee** [comment]
@oddlittlebird do you need help getting this to build?

**Comment by oddlittlebird** [comment]
> @oddlittlebird do you need help getting this to build?

Yes. That would be great.

**Comment by oddlittlebird** [comment]
NVM, I think I fixed them.

**Comment by oddlittlebird** [comment]
@daniellee , please take a look at this. I fixed the broken links, but now there is something else busted. Something about a YAML key.

**Comment by daniellee** [comment]
Bad indenting here: https://github.com/grafana/grafana/blob/8f460e4c49b2e47ff62906e83b1c72c50df82f62/docs/sources/menu.yaml#L214-L215

and a merge conflict too. Have pushed a fix for the indenting and hopefully I didn't mess up the merge conflict in the menu.yaml file.

File: docs/sources/linking/data-links.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+title = "Data links"
+keywords = ["grafana", "data links", "documentation", "playlist"]
+type = "docs"
++++
+
+# Data links
+
+Data links allow you to provide more granular context to your links. You can create links that include the series name or even the value under the cursor. For example, if your visualization showed four servers, you could add a data link to one or two of them.
+
+Click directly on the panel to see the data link. It appears on the context menu under **Add annotation**.
```
Comment (dprokop) [inline docs/sources/linking/data-links.md:11]: `It appears on the context menu under **Add annotation**.`

This is only true for Graph panel. You cannot add annotations o Stat, Gauge, Bar Gauge and Table which also support data links.

File: docs/sources/linking/data-links.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+The link itself is accessible in different ways depending on the visualization. For the graph you need to click on a data point or line, for a panel like
+Stat, Gauge, or Bar Gauge you can click anywhere on the visualization to open the context menu.
+
+>**Note:** For stat, gauge, bar gauge, and table visualizations, you add and edit data links on the Field tab. For the graph visualization, you add and edit data links on the Panel tab.
```
Comment (dprokop) [inline docs/sources/linking/data-links.md:16]: 👍

File: docs/sources/linking/data-links.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+## Add a data link
+
+1. Hover your cursor over the panel that you want to add a link to and then press `e`. Or click the dropdown arrow next to the panel title and then click **Edit**.
+1. On the Field tab, scroll down to the Data links section. (Panel tab for graph visualizations.)
+1. Expand Data links and then click **Add link**.
+1. Enter a **Title**. **Title** is a human-readable label for the link that will be displayed in the UI.
+1. Enter the **URL** you want to link to.
+   You can even add one of the template variables defined in the dashboard. Press Ctrl+Space or Cmd+Space and click in the **URL** field to see the available variables. By adding template variables to your panel link, the link sends the user to the right context, with the relevant variables already set. For more information, refer to [URL variables](../variables/url-variables.md)
```
Comment (dprokop) [inline docs/sources/linking/data-links.md:25]:
```suggestion
  Data links support multiple different variables, including template variables defined in your dashboard. Press Ctrl+Space or Cmd+Space and click in the **URL** field to see the available variables. By adding variables to your data link, the link sends the user to the right context, with the relevant variables already set. For more information, refer to [URL variables](../variables/url-variables.md)
```

**Review by dprokop — CHANGES_REQUESTED** [review]

File: docs/sources/linking/data-links.md (hunk 6 above)
Comment (oddlittlebird) [reply docs/sources/linking/data-links.md:11]: Then how do you access data links in the other panels?

**Review by dprokop — APPROVED** [review]

**Pull request merged**